piq/pr.py

```diff
@@ -0,0 +1,109 @@
+    """
+    if data_y is None:
+        data_y = data_x
+    dists = torch.cdist(data_x, data_y, p=2)
+    return dists
+
+
+def _get_kth_value(unsorted: torch.Tensor, k: int, axis: int = -1) -> torch.Tensor:
+    r"""
+    Args:
+        unsorted: Tensor of any dimensionality.
+        k: Int of the :math:`k`-th value to retrieve.
+    Returns:
+        kth values along the designated axis.
+    """
+    k_smallests = torch.topk(unsorted, k, dim=axis, largest=False)[0]
+    kth_values = k_smallests.max(dim=axis)[0]
+    return kth_values
+
+
+def _compute_nearest_neighbour_distances(input_features: torch.Tensor, nearest_k: int) -> torch.Tensor:
+    r"""Compute K-nearest neighbour distances.
+
+    Args:
+        input_features: Tensor of shape :math:`(N, feature_dim)`
+        nearest_k: Int of the :math:`k`-th nearest neighbour.
+    Returns:
+        Distances to :math:`k`-th nearest neighbours.
+    """
+    distances = _compute_pairwise_distance(input_features)
+    radii = _get_kth_value(distances, k=nearest_k + 1, axis=-1)
+    return radii
+
+
+class PR(BaseFeatureMetric):
+    r"""Interface of Improved Precision and Recall.
+    It's computed for a whole set of data and uses features from encoder instead of images itself to decrease
+    computation cost. Precision and Recall can compare two data distributions with different number of samples.
+    But dimensionalities should match, otherwise it won't be possible to correctly compute statistics.
+
+    Args:
+        real_features: Samples from data distribution. Shape :math:`(N_x, D)`
+        fake_features: Samples from generated distribution. Shape :math:`(N_y, D)`
+
+    Returns:
+        precision: Scalar value of the precision of image sets features.
+        recall: Scalar value of the recall of image sets features.
+
+    References:
+        Kynkäänniemi T. et al. (2019).
+        Improved Precision and Recall Metric for Assessing Generative Models.
+        Advances in Neural Information Processing Systems,
+        https://arxiv.org/abs/1904.06991
+    """
+
+    def __init__(self, nearest_k: int = 5) -> None:
+        r"""
+        Args:
+            nearest_k: Nearest neighbor to compute the non-parametric representation. Shape :math:`1`
+        """
+        super(PR, self).__init__()
+
+        self.nearest_k = nearest_k
+
+    def compute_metric(self, real_features: torch.Tensor, fake_features: torch.Tensor) \
+            -> Tuple[torch.Tensor, torch.Tensor]:
+        r"""Creates non-parametric representations of the manifolds of real and generated data and computes
+        the precision and recall between them.
+
+        Args:
+            real_features: Samples from data distribution. Shape :math:`(N_x, D)`
+            fake_features: Samples from fake distribution. Shape :math:`(N_x, D)`
+        Returns:
+            precision: Scalar value of the precision of the generated images.
+            recall: Scalar value of the recall of the generated images.
```

Comment:
The documentation parse the string before semicolon as a Return Type. I would suggest following.
```suggestion
            Scalar value of the precision of the generated images.
            
            Scalar value of the recall of the generated images.
```